### DISCUSSION TITLE
JS + Erlang actor library

### DISCUSSION THREAD

**Discussion by bobjoe12131** [gleam-lang/gleam#4104] [Ideas & suggestions]
In my current project, i realized that i need an mutable type, but if i used gleam/otp i would lose the JS target.
Is there anybody that is working on an actor library that supports Erlang and JS?
If not maybe i could take a crack at it, with some help and pointers because i have no idea how to make an actor framework.

**Comment by lpil** [comment]
It is impossible to abstract over the two different concurrency systems of JavaScript and Erlang as they have incompatible semantics.

**Reply by bobjoe12131** [reply]
Oh wait i forgot function calls happen sequentially as long as there isn't async, which my project doesn't have. I could just use a regular class and put the actions on two different targets. Ok, this was a dumb post, sorry for wasting your time.

**Reply by lpil** [reply]
Glad you figured it out!
